packages/react/src/SelectPanel/SelectPanel.tsx

```diff
@@ -141,7 +141,56 @@ const doesItemsIncludeItem = (items: ItemInput[], item: ItemInput) => {
   return items.some(i => areItemsEqual(i, item))
 }
 
-const defaultRendorAnchor: NonNullable<SelectPanelProps['renderAnchor']> = props => {
+interface KeyboardState {
+  height: number
+  isVisible: boolean
+}
+
+// calculate keyboard visibility state by detecting starting viewport height and subtracting viewport height when keyboard is visible
+function useKeyboardState(): KeyboardState {
+  const [keyboardState, setKeyboardState] = useState<KeyboardState>({
+    height: 0,
+    isVisible: false,
+  })
+
+  const initialHeightRef = useRef(0)
+
+  useEffect(() => {
+    if (window.visualViewport) {
+      initialHeightRef.current = window.visualViewport.height
+    }
+
+    const handleViewportChange = () => {
```

Comment:
@joshblack can we get an SSR-safe gut check here? not sure if this falls in the same categories as useMediaQuery, useResponsiveValues situation....

Reply:
I bet it's good since the select panel opens up only after an interaction, right?  So should be stable during SSR and only show up once client JS hydrates 👀

Reply:
yep, sounds good. Thanks!